samcli/hook_packages/terraform/hooks/prepare.py

```diff
@@ -51,25 +52,40 @@ def prepare(params: dict) -> dict:
         information of the generated metadata files
     """
     output_dir_path = params.get("OutputDirPath")
+    terraform_application_dir = params.get("IACProjectPath", os.getcwd())
     if not output_dir_path:
         raise PrepareHookException("OutputDirPath was not supplied")
 
     try:
         # initialize terraform application
         LOG.info("Initializing Terraform application")
-        run(["terraform", "init"], check=True, capture_output=True)
+        run(["terraform", "init"], check=True, capture_output=True, cwd=terraform_application_dir)
 
         # get json output of terraform plan
         LOG.info("Creating terraform plan and getting JSON output")
-        with NamedTemporaryFile() as temp_file:
-            run(["terraform", "plan", "-out", temp_file.name], check=True, capture_output=True)
-            result = run(["terraform", "show", "-json", temp_file.name], check=True, capture_output=True)
+
+        with osutils.tempfile_platform_independent() as temp_file:
+            run(
+                ["terraform", "plan", "-out", temp_file.name],
+                check=True,
+                capture_output=True,
+                cwd=terraform_application_dir,
+            )
+            result = run(
+                ["terraform", "show", "-json", temp_file.name],
+                check=True,
+                capture_output=True,
+                cwd=terraform_application_dir,
+            )
         tf_json = json.loads(result.stdout)
```

Comment:
we log it in case of failure .. https://github.com/moelasmar/aws-sam-cli/blob/180cc129e496d000771758a4a6734d9939277e7b/samcli/hook_packages/terraform/hooks/prepare.py#L99-L107
 
Also, we are going to enhance this part when we refactor the exception, and introduce a class that wrap Terraform